app/services/update_user.rb

```diff
@@ -46,12 +46,17 @@ def phone_attributes
   end
 
   def made_default_at_date
-    return current_made_default_at if attributes[:otp_make_default_number].blank?
-    Time.zone.now
+    otp_make_default_number = attributes[:otp_make_default_number]
+
+    if otp_make_default_number.blank? || otp_make_default_number == 'false'
+      current_made_default_at
+    else
+      Time.zone.now
+    end
   end
 
   def current_made_default_at
-    phone_configuration.made_default_at if attributes[:phone_id].present?
+    return phone_configuration.made_default_at if attributes[:phone_id].present?
     nil
```

Comment:
what if we removed both the `return` and the `nil` on the next line? 

```suggestion
    phone_configuration.made_default_at if attributes[:phone_id].present?
```